src/border_agent/border_agent.cpp

```diff
@@ -71,6 +71,12 @@
 #error "Border Agent feature requires at least one `OTBR_MDNS` implementation"
 #endif
 
+#define OT_THREAD_VERSION_1_4 5
+
+#ifndef OTBR_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ON_INIT
+#define OTBR_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ON_INIT (otThreadGetVersion() >= OT_THREAD_VERSION_1_4)
```

Comment:
I'd suggest to make this macro independent of runtime OT API. 

However, that's not achievable since there isn't a public OT macro defining the version. Let's continue discussion in https://github.com/openthread/ot-br-posix/pull/2429#discussion_r1713425771